archive_viewer/av_file_convert.py

```diff
@@ -247,12 +260,23 @@ def xml_to_dict(xml: ET.ElementTree) -> Dict:
         data_dict['plot_title'] = xml.find("plot_title").text
         data_dict['legend_configuration'] = xml.find("legend_configuration").attrib
 
-        for key in ("time_axis", "range_axis", "pv", "formula"):
+        for key in ("time_axis", "range_axis", "pv"):
             for element in xml.findall(key):
                 ele_dict = element.attrib
                 ele_dict |= {sub_ele.tag: sub_ele.text for sub_ele in element}
                 data_dict[key].append(ele_dict)
-
+        key = "formula"
+        for element in xml.findall(key):
+            ele_dict = element.attrib
+            curveDict = dict()
+            for sub_ele in element:
```

Comment:
O(n^2) here, it's probably fine. how many elements would one expect here?

Reply:
Technically it's O(n^2) but only if you have every formula referencing every PV. It's saying, in an xml file:

Find every formula. For each formula:

Do something to all of its sub_elements

Which means each attribute of each formula is looked at one time. If your formulas have n pvs that they reference, then sure it's n^2, but most likely one or two PVs are being referenced, in which case it's close to linear

Reply:
Maybe there would be a way to store a dictionary of old row names to new row names and convert them all at once so every *pv* is converted only once rather than every time in each formula, but I highly doubt this is necessary @zdomke and I discussed this as the way we would go about this solution. Either could make sense, and if you feel its a good change I can definitely add it in later!

Reply:
But for the record even if we were to make the xml to dict conversion O(n), the actual creation of the formulae on import time would definitely always require all of the referenced pvs to be referenced again, which means it will never avoid being "O(n^2)" in the exact same way

Reply:
this should be good! I wouldn't make changes for the time being